// pathpida/dist/buildTemplate.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.resetCache = void 0;
var path_1 = __importDefault(require("path"));
var createNextTemplate_1 = __importDefault(require("./createNextTemplate"));
var createNuxtTemplate_1 = __importDefault(require("./createNuxtTemplate"));
var createSapperTemplate_1 = __importDefault(require("./createSapperTemplate"));
var createStaticTemplate_1 = __importDefault(require("./createStaticTemplate"));
var prevPagesText = '';
var prevStaticText = '';
var resetCache = function () {
    prevPagesText = '';
    prevStaticText = '';
};
exports.resetCache = resetCache;
exports.default = (function (_a, mode) {
    var type = _a.type, input = _a.input, staticDir = _a.staticDir, output = _a.output, ignorePath = _a.ignorePath, trailingSlash = _a.trailingSlash, basepath = _a.basepath, pageExtensions = _a.pageExtensions;
    var emptyPathRegExp = /\n.+{\n+ +}.*/;
    if (mode !== 'static') {
        var text = {
            nextjs: function () { return (0, createNextTemplate_1.default)(input, output, ignorePath, pageExtensions); },
            nuxtjs: function () { return (0, createNuxtTemplate_1.default)(input, output, ignorePath, trailingSlash); },
            sapper: function () { return (0, createSapperTemplate_1.default)(input, output, ignorePath); }
        }[type]();
        while (emptyPathRegExp.test(text)) {
            text = text.replace(emptyPathRegExp, '');
        }
        prevPagesText = text;
    }
    if (staticDir && mode !== 'pages') {
        var text = (0, createStaticTemplate_1.default)(staticDir, basepath, ignorePath);
        while (emptyPathRegExp.test(text)) {
            text = text.replace(emptyPathRegExp, '');
        }
        prevStaticText = text;
    }
    return {
        text: "" + prevPagesText + prevStaticText + (type === 'nuxtjs'
            ? "\ndeclare module 'vue/types/vue' {\n  interface Vue {\n    $pagesPath: PagesPath" + (prevStaticText ? '\n    $staticPath: StaticPath' : '') + "\n  }\n}\n\ndeclare module '@nuxt/types' {\n  interface NuxtAppOptions {\n    $pagesPath: PagesPath" + (prevStaticText ? '\n    $staticPath: StaticPath' : '') + "\n  }\n\n  interface Context {\n    $pagesPath: PagesPath" + (prevStaticText ? '\n    $staticPath: StaticPath' : '') + "\n  }\n}\n\ndeclare module 'vuex/types/index' {\n  interface Store<S> {\n    $pagesPath: PagesPath" + (prevStaticText ? '\n    $staticPath: StaticPath' : '') + "\n  }\n}\n\nconst pathPlugin: Plugin = (_, inject) => {\n  inject('pagesPath', pagesPath)" + (prevStaticText ? "\n  inject('staticPath', staticPath)" : '') + "\n}\n\nexport default pathPlugin\n"
            : ''),
        filePath: path_1.default.posix.join(output, '$path.ts')
    };
});
